website/content/docs/concepts/cpu.mdx

```diff
@@ -0,0 +1,317 @@
+
+Modern CPUs may contain hetrogenous core types. Apple introduced the M1 CPU in
+2020 which contains both _performance_ (P-Core) and _efficiency_ (E-Core) types.
+Each core type operates a different base frequency. Intel introduced a similar
+topology in its Raptor Lake chips in 2022. When fingerprinting the characteristics
+of a CPU Nomad is capable of taking these advanced CPU topologies into account.
+
+[![PE Cores](/img/nomad-pe-cores.png)](/img/nomad-pe-cores.png)
+
+## Calculating CPU Resources
+
+The total CPU bandwidth of a Nomad node is the sum of the product between the 
+frequency of each core type and the total number of cores of that type in the
+CPU.
+
+```
+bandwidth = (p_cores * p_frequency) + (e_cores * e_frequency)
+```
+
+The total number of cores is computed by summing the number of P-Cores and the
+number of E-Cores.
+
+```
+cores = p_cores + e_cores
+```
+
+Nomad does not distinguish between logical and physical CPU cores. One of the
+defining differences between the P-Core and E-Core types is that the E-Cores
+do not support hyperthreading, wheras P-Cores do. As such a single physical
+P-Core is presented as 2 logical cores, and a single E-Core is presented as 1
+logical core.
+
+The example below is from a Nomad node with an Intel i9-13900 CPU. It is made
+up of mixed core types, with a P-Core base frequency of 2 GHz and an E-Core
+base frequency of 1.5 GHz.
+
+These characteristics are reflected in the `cpu.frequency.performance` and
+`cpu.frequency.efficiency` node attributes respectively. 
+
+```text
+cpu.arch                        = amd64
+cpu.frequency.efficiency        = 1500
+cpu.frequency.performance       = 2000
+cpu.modelname                   = 13th Gen Intel(R) Core(TM) i9-13900
+cpu.numcores                    = 32
+cpu.numcores.efficiency         = 16
+cpu.numcores.performance        = 16
+cpu.reservablecores             = 32
+cpu.totalcompute                = 56000
+cpu.usablecompute               = 56000
+```
+
+## Reserving CPU Resources
+
+In the fingerprinted node attributes, `cpu.totalcompute` indicates the total
+amount of CPU bandwidth the processor is capable of delivering. In some cases it
+may be beneficial to reserve some amount of a Node's CPU resources for use by
+the operating system and other non-Nomad processes. This can be done in client
+configuration.
+
+The amount of reserved CPU can be specified in bandwidth via `cpu`, e.g.
+
+```hcl
+client {
+  reserved {
+    cpu = 3000 # mhz
+  }
+}
+```
+
+Or the amount of reserved CPU can be specified by specifing a specific set of
+`cores` on which to disallow the scheduling of Nomad tasks. This capability is
+available on Linux systems only.
+
+```hcl
+client {
+  reserved {
+    cores = "0-3"
+  }
+}
+```
+
+Once the CPU is constrained by one of the above configurations, the node 
+attribute `cpu.usablecompute` will indicate the total amount of CPU bandwdith
+available for scheduling of Nomad tasks.
+
+## Allocating CPU Resources
```

Comment:
Something that feels missing from this doc is the mapping of scheduled CPU resources to what the task driver might (or might not!) do with that resource reservation. We have the `cpu_hard_limits` described below almost as an exception to the typical behavior. We should probably discuss that typical behavior in this section?

Reply:
Oh good call, added a blurb about isolation behavior